src/Stack/Config.hs

```diff
@@ -289,6 +296,11 @@ configFromConfigMonoid configStackRoot configUserConfigPath mresolver mproject c
          configAllowNewer = fromMaybe False configMonoidAllowNewer
          configDefaultTemplate = configMonoidDefaultTemplate
 
+     configAllowDifferentUser <-
+        case configMonoidAllowDifferentUser of
+            Just True -> return True
+            _ -> getInContainer
```

Comment:
This disables the ownership check inside containers as per @borsboom's [recommendation](https://github.com/commercialhaskell/stack/issues/1777#issuecomment-183122824).